### PR TITLE
Additions to the documentation and the testing

### PR DESCRIPTION
**Pull request by sbaldu**
List of changes:
* Updated and extended the documentation of the library in the README file. Now the doc contains a more thorough description of the entire library and all its methods
* Additions to the tests of the Python code (both for local testing and CI/CD)
* Additions to the testing datasets (both for local testing and CI/CD)
* Better error handling on the Python side
* Better use of numpy in the Python code for improved performance and readability

### CODE REVIEW

File: CLUEstering/CLUEstering.py [hunk 1]
```diff
@@ -47,53 +45,47 @@ def test_blobs(n_samples: int, n_dim: int , n_blobs: int = 4, mean: float = 0,
...
-
-            return pd.DataFrame(data)
-
-        # If it gets to the bottom we raise the exception
-        raise ValueError("Error: wrong number of dimensions\nBlobs can only be generated"
+    if x_max < 0. or y_max < 0.:
+        raise ValueError('Wrong parameter value. x_max and y_max must be positive.')
+    if n_blobs < 0:
+        raise ValueError('Wrong parameter value. The number of blobs must be positive.')
+    if mean < 0. or sigma < 0.:
```
Comment (waredjeb) [inline CLUEstering/CLUEstering.py:52]: maybe i am missing something, but why can't the `mean` be negative?

File: CLUEstering/CLUEstering.py [hunk 2]
```diff
@@ -290,78 +386,22 @@ def read_data(self,
...
-                self.scaler.fit_transform(self.clust_data.coords[dim].reshape(-1, 1)).reshape(1, -1)[0]
+        if isinstance(input_data, (str)):
+            df = self._read_string(input_data)
+            self._handle_dataframe(df)
+
+        if isinstance(input_data, (dict, pd.DataFrame)):
+            df = self._read_dict_df(input_data)
+            self._handle_dataframe(df)
+
+        # Rescae the coordinates with a standard scaler
```
Comment (waredjeb) [inline CLUEstering/CLUEstering.py:400]: typo: Rescae --> Rescale

File: CLUEstering/CLUEstering.py (hunk 1 above)
Comment (sbaldu) [reply CLUEstering/CLUEstering.py]: You're right

File: CLUEstering/CLUEstering.py [hunk 3]
```diff
@@ -215,36 +189,158 @@ class clusterer:
...
+        self.dc_ = dc_
+        self.rhoc = rhoc_
+        self.outlier = outlier_
+        self.ppbin = ppbin
 
         # Initialize attributes
         ## Data containers
-        self.clust_data = clustering_data()
+        self.clust_data = None
         self.scaler = StandardScaler()
```
Comment (waredjeb) [inline CLUEstering/CLUEstering.py:200]: do we want to let the user decide the Scaler? or maybe not use the scaler if the data have been already normalized?

File: CLUEstering/CLUEstering.py [hunk 4]
```diff
@@ -454,33 +494,29 @@ def choose_kernel(self,
...
+            if len(parameters) != 1:
+                raise ValueError("Wrong number of parameters. The flat kernel"
+                                 + " requires 1 parameter.")
+            self.kernel = Algo.flatKernel(parameters[0])
+        elif choice == "exp":
+            if len(parameters) != 2:
+                raise ValueError("Wrong number of parameters. The exponential"
+                                 + " kernel requires 2 parameters.")
+            self.kernel = Algo.exponentialKernel(parameters[0], parameters[1])
+        elif choice == "gaus":
```
Comment (waredjeb) [inline CLUEstering/CLUEstering.py:507]: gaus -> gauss ?

File: CLUEstering/CLUEstering.py (hunk 3 above)
Comment (sbaldu) [reply CLUEstering/CLUEstering.py:200]: I think that it's good to always scale,, also because it makes it easier to find the best combination of parameters.

Comment (waredjeb) [reply CLUEstering/CLUEstering.py:200]: I was thinking of avoiding scaling if already scaled to save some time, specially in case of big datasets. I guess we can leave the StandardScaler as default, but maybe the user can specify to "not scale", what do you think?

Comment (sbaldu) [reply CLUEstering/CLUEstering.py:200]: Ok makes sense. I'll add the option.
